crates/wasm-smith/src/lib.rs

```diff
@@ -2458,6 +2459,21 @@ struct Outer {
     modules: Vec<Rc<ModuleType>>,
 }
 
+/// A container for the kinds of instructions that wasm-smith is allowed to
+/// emit. Usage:
+/// ```
+/// # use wasm_smith::{InstructionKinds, InstructionKind};
+/// InstructionKinds::new(InstructionKind::Numeric | InstructionKind::Memory);
+/// ```
+#[derive(Clone, Copy, Debug, Default, Deserialize)]
```

Comment:
Is serde needed for this as a new dependency?

Reply:
Yes, unfortunately. The CLI and JSON configuration requires serialization and since this type is exposed there it must be `Deserialize`. I think we could avoid this and the `serde` import by doing something like  `pub fn new(kinds: Vec<InstructionKind>) -> Self`--I think serialization would then work and we could still get the benefit of a fast check during instruction generation.

Reply:
I think the JSON configuration would make more sense as `Option<Vec<String>>` and the CLI would be a comma-separated string, which would remove the need for serde here.

Reply:
`Option<Vec<String>>` makes it hard to fail when an invalid instruction kind is used. I kept it as an `Option<Vec<InstructionKind>>` to fail early on (instead of crashing in `allowed_instructions`) but then we can't get rid of the `serde` dependency. Is there somewhere else I can fail early?